DOCS-13161-mule-upgrade-tool-v4.4-cv (W-10590504)

Cristian-Venticinque opened the pull request:
Adding article for Mule upgrade tool.
Adding entry to nav.adoc.

Cristian-Venticinque committed:
Adding article for Mule upgrade tool.
Adding entry to nav.adoc.

Cristian-Venticinque committed:
Removed comments

Cristian-Venticinque committed:
Added `status` command description

Cristian-Venticinque committed:
Commented out feature not yet supported

Cristian-Venticinque committed:
Incorporated suggestions, minor fixes.

Cristian-Venticinque committed:
Fixed examples, added additional options for each command.

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+= Mule Upgrade Tool
```
Because there are task subtopics, we might need to retitle this as advised in our template, using a gerund title:
Upgrading Your Mule Instance? [Using the Mule Upgrade Tool]?

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+
+include::release-notes::partial$mule-upgrade-tool/mule-upgrade-tool.adoc[tag=intro]
+
+== Compatibility
```
How would you feel about using a more template-oriented approach to this topic, something like this:

= Upgrading Your Mule Instances by Using the Mule Upgrade Tool

Use the Mule upgrade tool to upgrade standalone instances of Mule 4. The tool enables you to upgrade local Mule instances to
//the latest available version or to
a particular Mule distribution that you have previously downloaded and unpacked), and to upgrade to Mule instances located in different folders. 

The Mule upgrade tool runs in any environment that is capable of running a standalone instance of Mule 4.

When the tool upgrades your current Mule instance, it also downloads the Runtime Manager agent that is bundled with the upgrade. If the Runtime Manager agent is already installed in your current Mule instance, the tool also updates the agent. 
[Question: if RTA is bundled with the upgrade, doesn't it stand to reason that the existing RTA is updated as well? Do we actually need to say this?]

The Mule upgrade tool also includes backup and rollback functionality.

 == Before You Begin....

 [[TBD?]] == Upgrade Your Mule 4 Instance Locally....steps....

 == Upgrade Your Mule 4 Instance to a Specific Distribution....steps....

 == Roll Back the Upgrade....steps

== Tool Commands and Options....reference

kbostrom replied:
I suggest this only because I found breaking up the introduction to the tool into smaller topics addressing compatibility, location of upgrade, and "considerations" a bit distracting. I think that task topic sets optimally comprise...
* An intro (concepts): why you perform the task and what it involves
* Prerequisites (if any)
* Steps

kbostrom replied:
I've also copyedited the existing material as-is in case you disagree with my suggestion. ;-)

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+include::release-notes::partial$mule-upgrade-tool/mule-upgrade-tool.adoc[tag=intro]
+
+== Compatibility
+
```
I'm not sure that these requirements need to be called out separately when they could be part of an introductory text, but if you do keep this list, it would be nice to have a text intro to it, following the subtitle and before the first list item.

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+
+== Before You Begin
+
+Before getting started, ensure you have:
```
```suggestion
Before starting your upgrade, ensure that you have:
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+* A Mule 4 instance currently configured and working in your local environment.
+* The Mule runtime engine version you want to upgrade to.
++
+You can download Mule from the https://help.mulesoft.com/s/[Support Portal^].
```
```suggestion
You can download this version from the https://help.mulesoft.com/s/[Support Portal^].
```

kbostrom replied:
(I don't know about your audience, but I always get confused that Mule 3 or Mule 4, which sound like versions of Mule to me, also have versions....)

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+* The Mule runtime engine version you want to upgrade to.
++
+You can download Mule from the https://help.mulesoft.com/s/[Support Portal^].
+* At least 2 GiB of available disk space on the file system and access privileges to install the new Mule distribution.
```
Use GB for gigabytes
Gb for gigabits

kbostrom replied:
(I like the phrase "the new Mule distribution"; it is less confusing to me than "the new Mule version")

kbostrom replied:
```suggestion
* At least 2 GB of available disk space on the file system and access privileges to install the new Mule distribution
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+
+Before getting started, ensure you have:
+
+* A Mule 4 instance currently configured and working in your local environment.
```
```suggestion
* A currently operational Mule 4 instance, in the stopped state to prepare for the upgrade
```
See Line 22 comment below.

kbostrom replied:
I would move the last bullet to follow this one, since they both address your current installation:

* The Mule upgrade tool inside your current Mule instance's `$MULE_HOME/tools` directory.

If the Mule upgrade tool is not bundled with your current Mule distribution, download the tool from the https://help.mulesoft.com/s/[Support Portal^] and copy it into the `$MULE_HOME/tools` directory of your current Mule instance.

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
++
+You can download Mule from the https://help.mulesoft.com/s/[Support Portal^].
+* At least 2 GiB of available disk space on the file system and access privileges to install the new Mule distribution.
+* The Mule instance to update in stopped status.
```
I would combine this with the first list item, because it is not quite parallel to the peer list items.

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+You can download Mule from the https://help.mulesoft.com/s/[Support Portal^].
+* At least 2 GiB of available disk space on the file system and access privileges to install the new Mule distribution.
+* The Mule instance to update in stopped status.
+//Confirm if the tool is downloaded from the Support portal or help center, and update the link.
```
```suggestion
//Confirm that the tool is downloaded from the Support portal or help center, and update the link.
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
++
+If the Mule upgrade tool is not bundled with your current Mule distribution, download the tool from the https://help.mulesoft.com/s/[Support Portal^] and copy it into the `$MULE_HOME/tools` directory of your current Mule instance.
+
+== Upgrade a Mule Instance
```
This title erroneously made me think that we had arrived at the steps to upgrade, but it's just more background/context/conceptual info that I would move to the topic set introduction, as shown earlier.
```suggestion
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+//the latest available version or to
+a particular Mule distribution that you have previously downloaded and unpacked. You can also use the Mule upgrade tool to update Mule instances located in different folders.
+
+=== Considerations
```
A vague title; suggest incorporating into intro.
```suggestion
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+
+=== Considerations
+
+When the tool upgrades your Mule instance, it also downloads the Runtime Manager agent that is bundled with the new Mule version. If the Runtime Manager agent was installed in your previous Mule instance, the tool also updates the agent to the new version that comes bundled with the new Mule version.
```
```suggestion
When the tool upgrades your current Mule instance, it also downloads the Runtime Manager agent that is bundled with the new Mule distribution. If the Runtime Manager agent is already installed in your current Mule instance, the tool updates the agent as part of the distribution.
```
Do we really need to point this out? Seems to me that if you upgrade the existing instance, every part of that instance is upgraded to every part of the new distribution. 
If we have to mention this at all, it might just be simpler to say that upgrading your instance includes updating or adding the Runtime Manager agent bundled as part of that instance.

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+
+//This feature is not yet available
+////
+=== Upgrade Mule to the Latest Version
```
This would become == if you omit the previous Upgrade title.
And is it accurate to use the parallel titles of upgrading locally vs. upgrading to a specific distribution?

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
````diff
@@ -0,0 +1,135 @@
+. Navigate to the Mule upgrade tool directory.
+. Run the `rollback` command:
+```
+$MULE_HOME/tools/upgrade-tool rollback
````
```suggestion
== Roll Back an Upgrade

The Mule upgrade tool generates a backup of your existing Mule instance before starting the upgrade process. The backup is stored in the `$MULE_HOME/.backup` folder in case you want to use the tool to restore your upgraded Mule instance to its previous state.

To perform a rollback:

. Navigate to the Mule upgrade tool directory.
. Run the `rollback` command:
```
$MULE_HOME/tools/upgrade-tool rollback
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+
+== Mule Upgrade Tool Commands and Options
+
+The following reference table shows a list of possible commands and options you can use when running the Mule upgrade tool:
```
```suggestion
The following table lists and describes commands and options you can use when running the Mule upgrade tool:
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+== Before You Begin
+
+Before getting started, ensure you have:
+
```
Remove periods from the ends of these list items.

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+Before getting started, ensure you have:
+
+* A Mule 4 instance currently configured and working in your local environment.
+* The Mule runtime engine version you want to upgrade to.
```
```suggestion
* The Mule runtime engine version you want to upgrade to
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+* At least 2 GiB of available disk space on the file system and access privileges to install the new Mule distribution.
+* The Mule instance to update in stopped status.
+//Confirm if the tool is downloaded from the Support portal or help center, and update the link.
+* The Mule upgrade tool inside your current Mule instance's `$MULE_HOME/tools` directory.
++
+If the Mule upgrade tool is not bundled with your current Mule distribution, download the tool from the https://help.mulesoft.com/s/[Support Portal^] and copy it into the `$MULE_HOME/tools` directory of your current Mule instance.
```
I would move this to follow the first list item.
```suggestion
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+
+== Upgrade a Mule Instance
+
+You can upgrade local Mule instances to
+//the latest available version or to
+a particular Mule distribution that you have previously downloaded and unpacked. You can also use the Mule upgrade tool to update Mule instances located in different folders.
```
ditto
```suggestion
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+a particular Mule distribution that you have previously downloaded and unpacked. You can also use the Mule upgrade tool to update Mule instances located in different folders.
+
+=== Considerations
+
```
Suggested final end to topic set intro, but evaluated for necessity:

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+!===
+  ! Options ! Description
+  ! `-n <newMuleDistributionPath>`
+  ! Upgrades the local Mule instance to a specific distribution that is downloaded and unpacked in your file system. Parameter `<newMuleDistribution>` specifies the path to the new Mule distribution.
+  +
+  For example:
```
```suggestion
  ! Upgrades the local Mule instance to a specific distribution that is downloaded and unpacked in your file system. Parameter `<newMuleDistribution>` specifies the path to the new Mule distribution, for example:
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+  ! `-o <localMuleInstancePath>`
+  ! Upgrades the local Mule instance located in path `<localMuleInstancePath>`. This option enables you to upgrade a different Mule instance than the one located in the same `$MULE_HOME` path as the upgrade tool.
+  +
+  For example:
```
```suggestion
  ! Upgrades the local Mule instance located in path `<localMuleInstancePath>`. This option enables you to upgrade a different Mule instance than the one located in the same `$MULE_HOME` path as the upgrade tool, for example:
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+!===
+  ! Options ! Description
+  ! `-o <localMuleInstancePath>`
+  ! Restores the local Mule instance located in path `<localMuleInstancePath>`. This option enables you to restore a different Mule instance than the one located in the same `$MULE_HOME` path as the upgrade tool.
+  +
+  For example:
```
ditto: GLOBALLY avoid putting "For example:" on a line by itself, as the sole introduction to an example or other nonnarrative element.
```suggestion
  ! Restores the local Mule instance located in path `<localMuleInstancePath>`. This option enables you to restore a different Mule instance than the one located in the same `$MULE_HOME` path as the upgrade tool, for example:
```

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+| Provides information about the current Mule instance, including:
+
+* The Mule runtime engine version
+* A message confirming if the current Mule instance mets conditions to upgrade or rollback
```
```suggestion
* A message confirming that the current Mule instance meets conditions to upgrade or roll back
```
rollback = adjective or noun
(to) roll back = verb

kbostrom commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,135 @@
+* The Mule runtime engine version
+* A message confirming if the current Mule instance mets conditions to upgrade or rollback
+* Any existing backup and its location
+* Integrity of the backup, if it exists
```
```suggestion
* The integrity of the backup, if it exists
```

kbostrom reviewed (dismissed):
CE with some CU thrown in for optional consideration

Cristian-Venticinque committed:
Editorial updates.

Cristian-Venticinque committed:
minor fixes

aiannucci commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,116 @@
+  ! `-o <localMuleInstancePath>`
+  ! Restores the local Mule instance located in path `<localMuleInstancePath>`. This option enables you to restore a different Mule instance than the one located in the same `$MULE_HOME` path as the upgrade tool, for example:
+  +
+  `$ ./upgrade-tool upgrade -o /<localMuleInstancePath>`
```
```suggestion
  `$ ./upgrade-tool rollback -o /<localMuleInstancePath>`
```

aiannucci commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,116 @@
+* The Mule upgrade tool inside your current Mule instance's `$MULE_HOME/tools` directory
++
+If the Mule upgrade tool is not bundled with your current Mule distribution, download the tool from the https://help.mulesoft.com/s/[Support Portal^] and copy it into the `$MULE_HOME/tools` directory of your current Mule instance.
+* The new Mule runtime engine distribution you want to upgrade to
```
I'm not sure if we should refer to "the distribution you want to upgrade to" or just refer to "the latest distribution", as the scope and aim of the upgrade tool is that you upgrade to the latest one. @estebanwasinger what do you think?

estebanwasinger replied:
+1

Cristian-Venticinque replied:
Addressed in [d6fd62a](https://github.com/mulesoft/docs-mule-runtime/pull/2159/commits/d6fd62ae387578659d4972550cd9f16a41264be1)

aiannucci commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
````diff
@@ -0,0 +1,116 @@
+```
+////
+
+== Upgrade Mule to a Specific Distribution
+
+To upgrade your Mule instance to a specific Mule distribution:
````
Same here, we might document this as the need to specify the path to where the new distribution is located, but not aimed to upgrade to a "specific" distribution. Although it is possible, we should encourage latest only.
cc @estebanwasinger

estebanwasinger replied:
Yes, despite that we are not doing any kind of version control, I would rephrase it to make it clear that you should always update to the "latest version"

Cristian-Venticinque replied:
Addressed in [d6fd62a](https://github.com/mulesoft/docs-mule-runtime/pull/2159/commits/d6fd62ae387578659d4972550cd9f16a41264be1)

Cristian-Venticinque committed:
Fixed requested changes.

anosenzo commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,117 @@
+[%header%autowidth.spread,cols=".^a,.^a"]
+!===
+  ! Options ! Description
+  ! `-n <newMuleDistributionPath>`
```
We could show the long option of the parameters and the help parameter as it is shown in the upgrade tool readme https://github.com/mulesoft/mule-runtime-upgrade-tool/blob/master/README.md#usage

anosenzo commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,117 @@
+* A message confirming that the current Mule instance meets conditions to upgrade or roll back
+* Any existing backup and its location
+* The integrity of the backup, if it exists
+
```
Should we add a troubleshooting section like in the readme? https://github.com/mulesoft/mule-runtime-upgrade-tool/blob/master/README.md#troubleshooting

Cristian-Venticinque committed:
Added info about solving conflicts with config files.

Cristian-Venticinque committed:
Added -help parameter

Cristian-Venticinque committed:
Added limitation about clusters.

Added clarification about using full distributions, not patches. 
Clarified the `upgrade -n` command.

Cristian-Venticinque committed:
Added prerequisite for Windows envs. 

Updated Help Center name in links.

damiancinich committed:
Rewrote section about solving conflicts

Cristian-Venticinque committed:
Added troubleshooting for file conflicts section

Cristian-Venticinque committed:
Added info for configuring logs.

Formatted code blocks.

damiancinich committed:
Added documentation about common errors

Cristian-Venticinque committed:
Reviewed and edited errors section. 


Reviewed wording to sync in terminology around commands and subcommands.

Cristian-Venticinque committed:
Fixed some punctuation and wording.

Cristian-Venticinque committed:
minor fix

Cristian-Venticinque committed:
Merge pull request #2201 from mulesoft/W-11019912

W 11019912 - Update Mule upgrade tool docs

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+
+include::release-notes::partial$mule-upgrade-tool/mule-upgrade-tool.adoc[tag=intro]
+
+The Mule upgrade tool runs in any environment that is capable of running a standalone Mule instance. This tool can update only standalone Mule instances that are running any Mule 4 version.
```
Mule Runtime Upgrade tool provides a fully automated solution for patching and upgrading standalone on-premise installations of Mule Runtime, version 4.0 or above. The tool is available as a separate download and can be used with existing or new Mule Runtime standalone deployments. The tool is available as command line utility so it can be easily integrated with existing automated processes or workflows via scripts and cron jobs.

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+
+The Mule upgrade tool runs in any environment that is capable of running a standalone Mule instance. This tool can update only standalone Mule instances that are running any Mule 4 version.
+//You can upgrade local Mule instances to the latest available version or to a particular Mule distribution that you have previously downloaded and unpacked.
+You can also use the Mule upgrade tool to update Mule instances located in different folders.
```
Remove the term "also". Proposed text here:

You can use the Mule upgrade tool to update Mule instances located in different folders.

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+//You can upgrade local Mule instances to the latest available version or to a particular Mule distribution that you have previously downloaded and unpacked.
+You can also use the Mule upgrade tool to update Mule instances located in different folders.
+
+As part of the upgrade process, the Mule upgrade tool also updates the Runtime Manager agent version if it is installed in your current Mule instance.
```
Note: 
1. Mule Runtime Upgrade tool is only available for Mule Runtime version 4 and above. 
2. If Runtime Manager agent is installed in your current Mule instance, Mule upgrade tool will upgrade the Runtime Manager agent version also. 
3. For clustered instances of Mule Runtime, you need to upgrade each cluster nodes independently by following the instructions in xref:release-notes::mule-runtime/updating-mule-4-versions.adoc#mulerunvers[Upgrading an On-Premises Mule Instance Managed Through Runtime Manager]

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+
+As part of the upgrade process, the Mule upgrade tool also updates the Runtime Manager agent version if it is installed in your current Mule instance.
+
+The Mule upgrade tool does not support upgrading clustered Mule instances. For clustered environments, upgrade each of the nodes manually by following the instructions in xref:release-notes::mule-runtime/updating-mule-4-versions.adoc#mulerunvers[Upgrading an On-Premises Mule Instance Managed Through Runtime Manager]
```
Remove this line. Covered in the note above.

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+
+Before starting your upgrade, ensure that you have:
+
+* A currently operational Mule 4 instance, in _stopped_ status to prepare for the upgrade
```
Mule Runtime version 4 or above

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+* A currently operational Mule 4 instance, in _stopped_ status to prepare for the upgrade
+* The Mule upgrade tool inside your current Mule instance's `$MULE_HOME/tools` directory
++
+If the Mule upgrade tool is not bundled with your current Mule distribution, download the tool from the https://help.mulesoft.com/s/[Help Center^] and copy it into the `$MULE_HOME/tools` directory of your current Mule instance.
```
Remove the line that it is not bundled with your Mule distribution. Suggested text: 

Download the Mule Runtime Upgrade tool from the https://help.mulesoft.com/s/[Help Center^] and copy it into the `$MULE_HOME/tools` directory of your current Mule instance.

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+If the Mule upgrade tool is not bundled with your current Mule distribution, download the tool from the https://help.mulesoft.com/s/[Help Center^] and copy it into the `$MULE_HOME/tools` directory of your current Mule instance.
+* The latest Mule runtime engine distribution, downloaded and unpacked in your system
++
+The Mule upgrade tool does not support upgrading to patch distributions, identified by including the word `patch` in the file name. Ensure you download a full Mule distribution, identified by including the word `distribution` in the file name, for example: `mule-ee-distribution-standalone-4.4.0-20220221.zip`.
```
The Mule Runtime upgrade tool requires full distribution of the newer patch or upgrade version of the Mule Runtime. Ensure you download a full Mule Runtime distribution identified by including the term `distribution` (and not the term 'patch') in the file name, for example: `mule-ee-distribution-standalone-4.4.0-20220221.zip`. 

@Cristian-Venticinque

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
++
+The Mule upgrade tool does not support upgrading to patch distributions, identified by including the word `patch` in the file name. Ensure you download a full Mule distribution, identified by including the word `distribution` in the file name, for example: `mule-ee-distribution-standalone-4.4.0-20220221.zip`.
++
+You can download Mule distributions from the https://help.mulesoft.com/s/[Help Center^].
```
"Mule Runtime"

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
````diff
@@ -0,0 +1,353 @@
+```
+////
+
+== Upgrade Mule
````
Patch or upgrade Mule Runtime to a new version

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+////
+== Upgrade Mule to the Latest Version
+
+To upgrade your Mule instance to the latest version:
```
"Mule Runtime instance"

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+
+To upgrade your Mule instance to the latest version:
+
+. Navigate to the Mule upgrade tool directory.
```
@Cristian-Venticinque -> please clarify which directory is this. Is this the directory where Upgrade tool files are extracted?

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+To upgrade your Mule instance to the latest version:
+
+. Navigate to the Mule upgrade tool directory.
+. Run the `upgrade -n` subcommand, specifying as an argument the directory where the latest Mule distribution is downloaded and unpacked:
```
"Mule Runtime"

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+
+== Roll Back an Upgrade
+
+The Mule upgrade tool generates a backup of your existing Mule instance before starting the upgrade process. The backup is stored in the `$MULE_HOME/backup` folder in case you want to use the tool to restore your upgraded Mule instance to its previous state.
```
Suggested text:

Before starting the upgrade, Mule upgrade tool creates a backup of your existing Mule Runtime instance. This is to let you restore your Mule Runtime instance to its previous state. This  backup gets stored in the `$MULE_HOME/backup` folder. "Mule Runtime instance"

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+
+To perform a rollback:
+
+. Navigate to the Mule upgrade tool directory.
```
"Mule Runtime" 
Also, please clarify which directory.

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+
+== Troubleshooting File Conflicts
+
+Before performing the upgrade, the tool compares configuration files from the _current_ Mule distribution and the _new_ distribution to detect changes and prevent these changes from being lost.
```
"the tool compares configuration files of the _current_ Mule Runtime distribution with the _new_ distribution to detect changes and prevent these changes from being lost.

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+. In the target environment, copy the files from `${MULE_HOME_WITH_USER_INTERACTION}/upgrade-tool/mule-configs-files/conf/` to `${MULE_HOME_WITHOUT_USER_INTERACTION}/upgrade-tool/mule-configs-files/conf/`.
+. Run the `upgrade` subcommand in the target environment.
+
+== Configure Logging for the Mule Upgrade Tool
```
IMO, this section should come before the troubleshooting section i.e. "Troubleshooting File Conflicts".

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+$MULE_HOME/tools/upgrade-tool rollback
+----
+
+== Troubleshooting File Conflicts
```
@Cristian-Venticinque troubleshooting section should come towards the end. "Configure Logging for the Mule Upgrade Tool" and "Mule Upgrade Tool Subcommands and Options" should come before this. @aiannucci - do you agree?

Cristian-Venticinque replied:
Originally, this was a subsection of the "Upgrade" command, since it is during this step where these errors arise, but I decided to leave it as an H2 header so it shows in the right navigation panel. 
Maybe it makes more sense to include it again as a sub-section of the upgrade command and rename it to "Solving File Conflicts"?

tanumutreja commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,353 @@
+
+=== Unattended Upgrades
+
+You can perform an unattended upgrade by resolving the conflicts in a different environment before running the upgrade in the target environment. This procedure works only if the _current_ and _new_ Mule distributions are in the same version in both environments:
```
@Cristian-Venticinque @aiannucci I don't really understand this "This procedure works only if the _current_ and _new_ Mule distributions are in the same version in both environments:" 

Can you make it clearer?

Cristian-Venticinque replied:
Would this help clarify: 
"This procedure works only if the _current_ and _new_ Mule distributions that you configure in the secondary environment match the versions of the _current_ and _new_ Mule distributions, respectively, that you have in the main environment:"

tanumutreja reviewed (commented):
Left feedback on the content and structure. @Cristian-Venticinque let me know when these changes are in so I can do a final review and sign-off.

Cristian-Venticinque committed:
Incorporated suggestions.

dukesphere commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,350 @@
+Download the tool from the https://help.mulesoft.com/s/[Help Center^] and copy it into the `$MULE_HOME/tools` directory of your current Mule instance.
+* The latest Mule runtime engine distribution, downloaded and unpacked in your system
++
+The Mule upgrade tool requires the full distribution of the latest Mule runtime minor or patch version. Ensure you download a full Mule runtime distribution, identified by including the term `distribution` (and not the term `patch`) in the file name, for example: `mule-ee-distribution-standalone-4.4.0-20220221.zip`.
```
Since it's sort of confusing that the full version of the patch update doesn't contain the term patch, I'm wondering if we need to indicate somehow that mule-ee-distribution-standalone-4.4.0-20220221.zip  is a full distribution of the patch update.

dukesphere replied:
maybe "... for example, for the full distribution of the 20220221 patch for Mule 4.4.0:"

dukesphere commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,350 @@
+----
+
+== Roll Back an Upgrade or Update
+Before starting the upgrade or update, Mule upgrade tool creates a backup of your existing Mule Runtime instance. This is to let you restore your Mule Runtime instance to its previous state. This backup gets stored in the $MULE_HOME/backup folder. "Mule Runtime instance"
```
```suggestion
```

dukesphere replied:
Looks like the edited version of the above paragraph is below.

dukesphere commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,350 @@
+
+You can use the Mule upgrade tool to update or upgrade Mule instances located in different folders.
+
+If the Runtime Manager agent is installed in your current Mule instance, the upgrade tool also updates the agent version as part of the upgrade process.
```
Wondering if we use "the" before "Runtime manager agent" and "Mule upgrade tool"?  Maybe yes if they are not official product names?

dukesphere commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,350 @@
++
+`${MULE_HOME}/upgrade-tool/mule-configs-files/ORIGINAL_CONFIG_FILES_NEW_DISTRO/`
+
+If the Mule upgrade tool fails the upgrade process due to conflicts with the configuration files, follow the next steps to resolve the conflicts and continue the upgrade process.
```
```suggestion
If the upgrade process fails due to conflicts with the configuration files, follow the next steps to resolve the conflicts and continue the upgrade process.
```

dukesphere commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,350 @@
+
+Follow these steps to resolve conflicts during an upgrade:
+
+. Inside the `${MULE_HOME}/upgrade-tool/mule-configs-files/conf/` folder of the current Mule distribution, locate one of the files that has a conflict.
```
"current" is probably understood but might be even more clear to say something like "of the current Mule distribution that you are upgrading" or some such

dukesphere commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,350 @@
+
+You can enable debug logging to add detailed information about each operation, change the folder where the tool stores the logs, and also use a customized log configuration file for the Mule upgrade tool.
+
+=== Specify a Custom Log4j File
```
```suggestion
=== Specify a Custom Log4j 2 File
```

dukesphere replied:
not sure if the "2" matters here and when talking about the related files, but just in case: https://logging.apache.org/log4j/2.x/

dukesphere commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,350 @@
+
+== Mule Upgrade Tool Subcommands and Options
+
+The following table lists and describes the subcommands and options you can use when running the Mule upgrade tool:
```
```suggestion
The following table describes the subcommands and options you can use when running the Mule upgrade tool:
```

dukesphere replied:
I should've noticed that the first time I looked at that sentence... oh well.

dukesphere commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,350 @@
+The Mule upgrade tool requires the full distribution of the latest Mule runtime minor or patch version. Ensure you download a full Mule runtime distribution, identified by including the term `distribution` (and not the term `patch`) in the file name, for example: `mule-ee-distribution-standalone-4.4.0-20220221.zip`.
++
+You can download Mule runtime distributions from the https://help.mulesoft.com/s/[Help Center^].
+* At least 2 GB of available disk space on the file system and access privileges to install the new Mule distribution
```
Here we say 2 GB. Below we say 2GiB.

dukesphere commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,350 @@
+
+==== Solution
+
+The tool requires at least 2GiB of space available in the file system in which the Mule instance is running. Ensure that your system has
```
Compare 2GiB  here with a 2 GB above.

dukesphere reviewed (commented):
There was some essentially duplicated text. Otherwise, just some suggestions to consider.

Cristian-Venticinque committed:
Editorial review.

Cristian-Venticinque committed:
Merge branch 'v4.4' into DOCS-13161-mule-upgrade-tool-v4.4-cv

Cristian-Venticinque committed:
Updating directory paths in conflict solving section.

damiancinich committed:
Added cases to Common Errors

damiancinich committed:
Update modules/ROOT/pages/mule-upgrade-tool.adoc

damiancinich committed:
Update modules/ROOT/pages/mule-upgrade-tool.adoc

Co-authored-by: Duke <dukesphere@users.noreply.github.com>

damiancinich committed:
Update modules/ROOT/pages/mule-upgrade-tool.adoc

Co-authored-by: Duke <dukesphere@users.noreply.github.com>

damiancinich committed:
Update modules/ROOT/pages/mule-upgrade-tool.adoc

Co-authored-by: Duke <dukesphere@users.noreply.github.com>

damiancinich committed:
Update modules/ROOT/pages/mule-upgrade-tool.adoc

Co-authored-by: Duke <dukesphere@users.noreply.github.com>

damiancinich committed:
Update modules/ROOT/pages/mule-upgrade-tool.adoc

Co-authored-by: Duke <dukesphere@users.noreply.github.com>

damiancinich committed:
Update modules/ROOT/pages/mule-upgrade-tool.adoc

Co-authored-by: Duke <dukesphere@users.noreply.github.com>

dukesphere committed:
Merge pull request #2225 from mulesoft/W-11086312

W-11086312 and W-11174370: Added cases to Common Errors

Cristian-Venticinque committed:
Updated requirements to clarify supported Mule versions (4.1.5+)

dukesphere committed:
Merge branch 'v4.4' into DOCS-13161-mule-upgrade-tool-v4.4-cv

Prannjall11 reviewed (dismissed):
documentation PR looks good.

anosenzo commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,399 @@
+
+* Fails the execution of the `upgrade` subcommand, showing an error message
+* Logs the names of the Mule configuration files with conflicts into the log output
+* Copies the configuration files with conflicts from the `${MULE_HOME}/conf` directory of the current Mule distribution to `${MULE_HOME}/upgrade-tool/mule-config-conflicts/to-review`.
```
```suggestion
* Copies the configuration files with conflicts from the `${MULE_HOME}/conf` directory of the current Mule distribution to `${MULE_HOME}/upgrade-tool/mule-config-conflicts/to_review`.
```

anosenzo commented on modules/ROOT/pages/mule-upgrade-tool.adoc:
```diff
@@ -0,0 +1,399 @@
+==== Solution
+
+Stop Mule before starting the upgrade process. To check the current status, use the command `${MULE_HOME}/bin/mule status`.
+
```
We could add this troubleshooting section: https://github.com/mulesoft/mule-runtime-upgrade-tool#your-mule-version-is-not-supported-for-an-upgrade

Cristian-Venticinque committed:
Fixed folder path. 
Added new error cause and solution.

Cristian-Venticinque committed:
Updated example to sync with other error messages
